### ISSUE TITLE
Datestamp in titles results in duplicate titles if taken on the same day

### ISSUE THREAD

**Issue by wresuolc** [mozilla-services/screenshots#4211]
From [discourse](https://discourse.mozilla.org/t/file-names-not-unique-on-download/26421/1):

> When I take multiple screenshots and save them to disk on the same site and on the same day, they produce the same file name (example: ‘Screenshot-2018-3-8 GeoGuessr - Let’s explore the world .png’). I then have to manually edit one of the titles before I am allowed to save the screenshots in the same directory.
> 
> I propose adding a nonce to the title to solve the issue.

**Comment by wresuolc** [comment]
This was reported as producing the same name, but testing shows the filenames have a number in them, eg:  file.png -> file(1).png -> file(2).png, etc.  I think it's put there by the browser?  So...we're going to close this.

**Comment by gentoo-root** [comment]
I'm on the latest Firefox 59.0.2, GNOME 3, and I'm experiencing the duplicate filenames issue.

Steps to reproduce:
1. Visit a website (e.g. this page on github).
2. Take a screenshot of the visible part of the page.
3. The save dialog appears with the Downloads directory open in it.
4. Change the directory e.g. to Pictures.
5. Save the screenshot.
6. Take another screenshot of the visible part of the same page.
7. The save dialog appears, and it shows the Downloads directory again. As there is no file with this name in this directory, "(1)" is not appended.
8. Change directory to Pictures.
9. Try to save the screenshot, and the save dialog warns you that you are trying to overwrite the existing file.

Expected result:
No filename collision between the screenshots taken at different times.

Actual result:
Screenshots taken at different times have the same filename, and the user is forced to manually rename the file.

Additional comments:
"(1)", "(2)", etc. are really appended to the filename if you don't change the directory in the save dialog and just save everything to Downloads. But if you want to store the screenshots to a different directory, you are forced to select it manually each time which is annoying, and no "(1)", "(2)", etc. are appended in this case. The best solution in my opinion is to just include a timestamp (including seconds) in the filename.

**Comment by 6a68** [comment]
@gentoo-root Thanks for the feedback. This bug has been closed as a wontfix, sorry. 

FWIW, on linux, you should be able to use something like `inotify` to auto-detect filesystem changes, and automatically rename any files matching the Screenshots initial part to include a local timestamp. I realize this isn't the answer you're after, but at least you can easily hack together a DIY solution :+1:

**Comment by gentoo-root** [comment]
>This bug has been closed as a wontfix, sorry.

ORLY? I can't see any mention of this resolution ITT. What I see is the following:

>testing shows the filenames have a number in them, eg: file.png -> file(1).png -> file(2).png, etc. I think it's put there by the browser? So...we're going to close this.

which means that the bug is closed as invalid/not reproduced. I suggest the steps to reproduce the bug so that you all could see that it's valid. So why not just follow the steps, reproduce the bug and reopen it, then fix it?

It's trivial to fix that little tiny thing that irritates and annoys a lot of users, but instead you suggest some ugly workarounds when it's Firefox Screenshots' job to name the files correctly. Instead of creating the file with the correct name, you suggest creating a file with a wrong name, running a daemon that runs inotify watches and renames them. And I strongly believe that you understand the consequences of that "solution" — it's easy enough to get into an inconsistent state, be it a power loss before renaming, or a daemon crash, etc. There is a kind of people who write low-quality bash scripts for everything with hardcoded `DISPLAY=:0`, race conditions, improper escaping, etc. I'm not one of those people, and I'm not gonna turn my computer into a dump by writing buggy scripts when someone refuses to fix their buggy software.

It's too sad to admit that the new Firefox has taken a self-destructive course. Firefox Screenshots demonstrate that perfectly. First half of the extensions is broken including the extension I used to take screenshots. Then Mozilla introduces a built-in screenshotting tool which is, by the way, inferior to the one I used before. And what we see here? A big blue Save button that sends my private data to the cloud instead of saving my screenshots (#3603). Ugly error messages when just following a normal user flow (#3964). Too many clicks to just save a screenshot — it takes me at least 5 clicks just to save a shot of a page when it had taken me only a single click when I used a third-party screenshotter extension. One screenshot a day limit (#4229, #4211). No way to customize the filename pattern (#4229). Doesn't remember the directory where I store all my screenshots (#4211). These all make this piece of software absolutely unusable. These are the real problems, because they break UX. However, from version to version, nothing changes in Firefox Screenshots, still the same problems, still unusable. And the developers just say they are not going to fix that. Firefox Screenshots' fate is sealed. Just like Firefox Hello, it started as a new good useful tool, but it will end up exactly the same way, because it doesn't seem to be developed any more.

I beg my pardon for possibly rude voice, but I'm seriously concerned about my favorite browser's future, and such responses from the developers are no good sign.
